components/google_drive/actions/add-file-sharing-preference/add-file-sharing-preference.js

```diff
@@ -0,0 +1,96 @@
+const googleDrive = require("../../google_drive.app");
+
+module.exports = {
+  key: "google_drive-add-file-sharing-preference",
+  name: "Add File Sharing Preference",
+  description: "Add File Sharing Preference",
+  version: "0.0.12",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description: "The drive you want to find a file in",
+      optional: true,
+      default: "",
```

Comment:
I see, so if we default to `""`, that means we end up fetching data for the user's authenticated drive - is that right?

Would you mind taking a look at [Zapier](https://zapier.com/) to see how they implement? They have a solid user experience, so I'd like to see if they make that prop optional, as well.

@dannyroosevelt when we get to the QA stage with this we should talk about whether to make the drive optional or required. If it's not set, it defaults to your current drive. I think this behavior could be OK, it just looks a little odd in the current UI because of how we handle optional props.

Reply:
Yes, if we default to `""` or undefined, we fetch data for the user's authenticated drive, including files in the "Shared with me" category.

Reply:
I set a default of `""` because of the ESLint warning, and because `""` currently works the same as `undefined` because the `drive` prop is used in a loose equality check. But I'm guessing there's a better solution. I meant to leave a comment about defaults for optional props. 😅

Reply:
Thanks. I'll defer to @dannyroosevelt on the question of whether the prop should be optional (from a product perspective). I think the default listing is probably what most users want. The only issue from my perspective is that it's listed as an optional prop first, which our current UI doesn't handle well. But long-term keeping it optional is probably best.

Can you remove the `default: ""` here and anywhere where `undefined` will work as a default? I think I'll probably just remove the ESLint warning. In general I'll only add `default` when there's a non-empty value.

Reply:
Zapier makes the `drive` prop optional, and it's positioned as the first prop to configure. On Zapier, its description includes "If nothing is selected, then your personal Google Drive will be used."